src/LiveComponent/src/LiveComponentHydrator.php

```diff
@@ -145,7 +146,20 @@ public function hydrate(object $component, array $props, array $updatedProps, Li
         $attributes = new ComponentAttributes($dehydratedOriginalProps->getPropValue(self::ATTRIBUTES_KEY, []));
         $dehydratedOriginalProps->removePropValue(self::ATTRIBUTES_KEY);
 
-        foreach ($componentMetadata->getAllLivePropsMetadata() as $propMetadata) {
+        $allMetadata = $componentMetadata->getAllLivePropsMetadata();
+        uasort($allMetadata, static function (LivePropMetadata $a, LivePropMetadata $b) {
+            if ($a->hasModifier() && !$b->hasModifier()) {
+                return 1;
+            } elseif (!$a->hasModifier() && $b->hasModifier()) {
+                return -1;
+            } else {
+                return 0;
+            }
+        });
```

Comment:
Why do we need to put the method with a modifier in priority of others?

Reply:
Metadata with a modifier are considered "greater" than others, and the array is sorted in asc direction. I'd say this code prioritize metadata without modifier?